Sources/MoSoCore/PocketAuthenticationService.swift

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Should be removed once the Mastodon Token supports the Pocket API
+
+import Foundation
+import AuthenticationServices
+import Combine
+
+struct Response {
+    let guid: String
+    let accessToken: String
+    let userIdentifier: String
+}
+
+enum PocketAuthenticationError: Error {
+    case invalidRedirect
+    case invalidComponents
+    case alreadyAuthenticating
+    case noConsumerKey
+    case other(Swift.Error)
+}
+
+class PocketAuthenticationService: NSObject, ASWebAuthenticationPresentationContextProviding {
```

Comment:
Out of my curiosity, since this is part of `MoSoCore`, how does this work when the package is imported by a third party app (e. g. a Mastodon client) that provides its own authentication?

Reply:
This a temporary measure solely for the ReadingListKit auth.
It should be deleted once the Moz.Social auth token is usable in the Pocket API.

Reply:
> This a temporary measure solely for the ReadingListKit auth.

I get this. Do you think there could be a way to (quoting the requirements document) 

> [...] Leverage existing MoSoContentKit types and modules to handle tokens [...]

in order to make this library agnostic of the particular authentication implementation and, once the new Mozilla.Social auth is ready, just use that token instead of the one from the existing Pocket auth?

Reply:
If you compare PocketAuthorizationService to AuthorizationService you can see they are pretty different.
The Mastodon OAUTH flow is already server agnostic because they all conform to the same protocols.

Pocket OAUTH is fairly specific so I don't think it would be helpful to try merge it into a single OAUTH class. And since we are currently thinking it will be removed we would just have to undo that work at a later date. The current implementation is intended to be as easily removable as possible.

The auth tokens are already intended to be controlled by MoSoSesson, held in the Keychain. I've added functions to fetch the Mastodon token and the Pocket token. 

MoSoSession.swift (Line 24)
```
    public func authToken() -> String {
        "ACCESS_KEYCHAIN_HERE"
    }

    // Should be removed once the Mastodon Token supports Pocket API access
    public func pocketAuthToken() -> String {
        "ACCESS_KEYCHAIN_HERE"
    }
```

To make the switch over as painless as possible there is one place you would need to change which one is being used.

AppConfigurator.swift (Line 49)
OLD
```
    func makeReadingListTab() -> some View {
        ReadingListView(model: ReadingListModel(authTokenProvider: session.pocketAuthToken, consumerKey: Keys.shared.pocketAPIConsumerKey))
    }
```

NEW
```
    func makeReadingListTab() -> some View {
        ReadingListView(model: ReadingListModel(authTokenProvider: session.authToken, consumerKey: Keys.shared.pocketAPIConsumerKey))
    }
```
And that would be enough to have the ReadingList switch over to using the Mastodon auth token. We could even wrap that in a feature flag to enable remote switching and a slow rollout of the switch.

Does that clear everything up?
Happy to jump on a call and talk through alternatives.

Reply:
Thanks for the additional info @CMasterson !

> Pocket OAUTH is fairly specific so I don't think it would be helpful to try merge it into a single OAUTH class.

to clarify, when I say

> to make this library agnostic of the particular authentication implementation

I don't necessarily mean to merge classes, rather to make the library `agnostic` of the authentication itself.

> The auth tokens are already intended to be controlled by MoSoSesson, held in the Keychain

That's great, and it's another step towards the solution!

Reply:
ReadingListKit should be pretty agnostic at this point. All it cares about is that it's token is provided by a function with signature `() -> String`. We can firm that up into an Alias or something later.

> only one step remaining

Not even nearly!
We have to figure out storage of the token.
How we're going to fire the initial fetch.
Who is actually going to fire the network requests.
Then, depending on the result; 
- How we're going to handle refreshing the token.
- Error propagation

Reply:
Oh I meant one step to abstract the authentication away. 

> ReadingListKit should be pretty agnostic at this point.

Could you elaborate on this? It seems that the pocket auth still resides in it, but maybe you are talking about future evolutions of this PR?

Reply:
Pocket Auth is in MoSoCore
Sources/MoSoCore/PocketAuthenticationService.swift

I put it there to keep it close to MoSoSession.
Moving it out of ReadlingList was the right call, as we can have one interface and inject the appropriate token provider.

Reply:
Right, but `MoSoCore` it's still a library that we are going to distribute, isn't it?